apps/resource_manager/lib/credentials/schemas/public_key.ex

```diff
@@ -38,28 +39,25 @@ defmodule ResourceManager.Credentials.Schemas.PublicKey do
     timestamps()
   end
 
-  @doc false
-  def changeset_create(params) when is_map(params) do
-    %__MODULE__{}
-    |> cast(params, @required_fields ++ @foreign_key_fields ++ @optional_fields)
-    |> validate_required(@required_fields ++ @foreign_key_fields)
-    |> validate_inclusion(:type, @possible_types)
-    |> validate_inclusion(:format, @possible_formats)
-    |> unique_constraint(:user_id)
-  end
+  @doc "Generates an `%Ecto.Changeset{}` to be used in insert operations"
+  @spec changeset(params :: map()) :: Ecto.Changeset.t()
+  def changeset(params) when is_map(params), do: changeset(%__MODULE__{}, params)
```

Comment:
```suggestion
  def changeset(%{} = params), do: changeset(%__MODULE__{}, params)
```

Reply:
I prefer to use guard when it's an map or list and the content itself doesen't matter.
But this is just personal preference hahaha.